diff --git a/torrent.go b/torrent.go
index 4d36209..d8802cd 100644
--- a/torrent.go
+++ b/torrent.go
@@ -544,7 +544,41 @@ func (t *Torrent) numPieces() int {
 func (t *Torrent) numPiecesCompleted() (num int) {
 	return t.completedPieces.Len()
 }
+func (t *Torrent) Announce() {
+	t.wantPeersEvent.Set()
+	t.announceDHT(true)
+	t.startMissingTrackerScrapers()
+	cl := t.cl
+	cl.event.Broadcast()
+	cl.openNewConns(t)
+}
+func (t *Torrent) Reopen() {
+	t.closed.SetBool(false)
+	t.closed.Clear()
+	t.announceDHT(true)
+	t.wantPeersEvent.Set()
+	t.startMissingTrackerScrapers()
+	cl := t.cl
+	cl.event.Broadcast()
+	cl.openNewConns(t)
+	t.openNewConns()
+	t.publishPieceChange(0)
+}
+func (t *Torrent) Close() (err error) {
+
+	t.closed.Set()
+	if t.storage != nil {
+		t.storage.Close()
+	}
+	for conn := range t.conns {
+		conn.Close()
+	}
 
+	t.pieceStateChanges.Close()
+	t.updateWantPeersEvent()
+	return
+
+}
 func (t *Torrent) close() (err error) {
 	t.closed.Set()
 	if t.storage != nil {
